### PR TITLE
Allocated separate domains for endpoints in RDM tests

### LINKED ISSUE #523: Use different domains for each endpoint in the RDM tests
Allocate separate domains for each endpoint

### PR DESCRIPTION
**Pull request by e-harvey**
```
* Allocated separate domains for each endpoint in:
  rdm_atomic.c
  rdm_rma.c
  rdm_sr.c
```

Fixes #523
@ztiffany

### CODE REVIEW

File: prov/gni/test/rdm_rma.c [hunk 1]
```diff
@@ -574,19 +673,19 @@ void do_inject_write(int len)
...
-		}
+	  loops = 0;
+	  while (source[i] != target[i]) {
+	    ret = fi_cq_read(send_cq[0], &cqe, 1); /* for progress */
+	    cr_assert(ret == -EAGAIN,
+		      "Received unexpected event\n");
+
+	    pthread_yield();
+	    cr_assert(++loops < 10000, "Data mismatch");
+	  }
```
Comment (ztiffany) [inline prov/gni/test/rdm_rma.c]: You're using a different tab width here.

File: prov/gni/test/rdm_rma.c [hunk 2]
```diff
@@ -658,23 +760,23 @@ void do_inject_writedata(int len)
...
-		}
+			loops = 0;
+			while (source[i] != target[i]) {
+				ret = fi_cq_read(send_cq[0], &cqe, 1); /* for progress */
+				cr_assert(ret == -EAGAIN,
+					  "Received unexpected event\n");
+
+				pthread_yield();
+				cr_assert(++loops < 10000, "Data mismatch");
+			}
```
Comment (ztiffany) [inline prov/gni/test/rdm_rma.c]: Why was another indent added here?

File: prov/gni/test/rdm_sr.c [hunk 3]
```diff
@@ -1234,12 +1251,13 @@ void do_send_buf(void *s, void *t, int len)
 	} while (!(source_done && dest_done));
 
 	rdm_sr_check_cqe(&s_cqe, t, (FI_MSG|FI_SEND), 0, 0, 0);
-	rdm_sr_check_cqe(&d_cqe, s, (FI_MSG|FI_RECV), t, len, 0);
-	rdm_sr_check_cntrs(1, 1, 0, 0);
+	rdm_sr_check_cqe(&d_cqe, p, (FI_MSG|FI_RECV), t, len, 0);
+	s[0] = 1; r[1] = 1;
+        rdm_sr_check_cntrs(s, r, s_e, r_e);
```
Comment (ztiffany) [inline prov/gni/test/rdm_sr.c]: A lot of the rdm_sr_check_cntrs() lines in this file replaced a tab indent with spaces.

File: prov/gni/test/rdm_sr.c [hunk 4]
```diff
@@ -1205,20 +1221,21 @@ Test(rdm_sr, send_readfrom)
...
 }
 
-void do_send_buf(void *s, void *t, int len)
+void do_send_buf(void *p, void *t, int len)
 {
 	int ret;
 	int source_done = 0, dest_done = 0;
 	struct fi_cq_tagged_entry s_cqe, d_cqe;
 	ssize_t sz;
+	uint64_t s[NUMEPS] = {0}, r[NUMEPS] = {0}, s_e[NUMEPS] = {0}, r_e[NUMEPS] = {0};
```
Comment (ztiffany) [inline prov/gni/test/rdm_sr.c]: It looks like a lot of the edits you made throughout push lines past the 80 character limit.  You probably want to get the linux stye checker working to point out those issues when you commit.

File: prov/gni/test/rdm_sr.c [hunk 5]
```diff
@@ -894,38 +890,41 @@ void do_recvv(int len)
...
+
 }
 
 /*
-ssize_t (*recvmsg)(struct fid_ep *ep, const struct fi_msg *msg,
-		uint64_t flags);
- */
+  ssize_t (*recvmsg)(struct fid_ep *ep, const struct fi_msg *msg,
+  uint64_t flags);
+*/
```
Comment (ztiffany) [inline prov/gni/test/rdm_sr.c:920]: I don't understand why these prototype comments were edited.  Not a big deal but it makes the diff larger and changes from more standard prototype formatting.

File: prov/gni/test/rdm_rma.c (hunk 1 above)
Comment (e-harvey) [inline prov/gni/test/rdm_rma.c]: This is from not having my text editor configured properly on tiger.

File: prov/gni/test/rdm_rma.c (hunk 2 above)
Comment (e-harvey) [inline prov/gni/test/rdm_rma.c]: That was a mistake.

File: prov/gni/test/rdm_sr.c (hunk 3 above)
Comment (e-harvey) [inline prov/gni/test/rdm_sr.c]: This is from my text editor config on tiger.

File: prov/gni/test/rdm_sr.c (hunk 4 above)
Comment (e-harvey) [inline prov/gni/test/rdm_sr.c]: I will go through and split lines that are longer than 80 characters.

File: prov/gni/test/rdm_sr.c (hunk 5 above)
Comment (e-harvey) [inline prov/gni/test/rdm_sr.c:920]: Ok, they were edited in an effort to maintain the original format of the file.

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by hppritcha** [comment]
@ztiffany  are you good with this?

**Comment by ztiffany** [comment]
:+1:

**Pull request merged**